### ISSUE TITLE
Return value of fcntl not checked

### ISSUE THREAD

**Issue by tibbe** [nyholku/purejavacomm#88]
At https://github.com/nyholku/purejavacomm/blob/5301918f25ad73041d6fc261da13f14c602d274f/src/purejavacomm/PureJavaSerialPort.java#L1091 there's a call to fcntl which return value isn't checked:

```java
int flags = fcntl(m_FD, F_GETFL, 0);
```

Since this call can fail we might pass an invalid value for `flags` just after:

```java
flags &= ~O_NONBLOCK;
checkReturnCode(fcntl(m_FD, F_SETFL, flags));
```

This in turn can lead to an exception being raised, but at the wrong point, suggesting that what failed was the flag set, not the flag get!

**Comment by tibbe** [comment]
I ran into this using the Celestron Firmware Manager, which uses this library. Here's the full stack trace:

```
09:29:36.268 _003.924.860 _ java.util.concurrent.ExecutionException: purejavacomm.PureJavaIllegalStateException: JTermios call returned -1 at class purejavacomm.PureJavaSerialPort line 1093
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at cfm.devicetask.DeviceTask.get(DeviceTask.java:390)
        at cfm.devicetask.DeviceTask.done(DeviceTask.java:300)
        at cfm.devicetask.DeviceTask.access$000(DeviceTask.java:94)
        at cfm.devicetask.DeviceTask$2.done(DeviceTask.java:134)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:384)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:251)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at cfm.devicetask.DeviceTask.run(DeviceTask.java:380)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: purejavacomm.PureJavaIllegalStateException: JTermios call returned -1 at class purejavacomm.PureJavaSerialPort line 1093
        at purejavacomm.PureJavaSerialPort.checkReturnCode(PureJavaSerialPort.java:1294)
        at purejavacomm.PureJavaSerialPort.<init>(PureJavaSerialPort.java:1093)
        at purejavacomm.CommPortIdentifier.open(CommPortIdentifier.java:148)
        at cfm.link.SerialLink.acquireLinkInner(SerialLink.java:160)
        at cfm.link.SerialLink.acquireLink(SerialLink.java:126)
        at cfm.devicetask.SeekPrimaryTask.testPortal(SeekPrimaryTask.java:132)
        at cfm.devicetask.SeekPrimaryTask.doInBackground(SeekPrimaryTask.java:81)
        at cfm.devicetask.SeekPrimaryTask.doInBackground(SeekPrimaryTask.java:26)
        at cfm.devicetask.DeviceTask$1.call(DeviceTask.java:119)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        ... 4 more
```

It's hard to tell what the issue is as the raise on F_SETFL could have actually been triggered by the fcntl 2 lines above.

**Comment by nyholku** [comment]
Hi, thanks for the feedback. On cursory look I do not see that F_GETFL can fail and if it fails how can we check that it failed.

http://linux.die.net/man/2/fcntl

**Comment by tibbe** [comment]
It can fail with EBADF in case of bad file descriptor (http://www.gnu.org/software/libc/manual/html_node/Getting-File-Status-Flags.html). In that case it will return -1 and errno will be set.

**Comment by nyholku** [comment]
Ok, the man page then leaves something to desire for ;) ... ok I will fix this, thanks for spotting and contributing.

**Comment by nyholku** [comment]
Ok, this has been addressed will be released in 1.0.1
